core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -147,7 +147,13 @@ where
                             MAX_PROCESSING_AGE,
                         )
                     },
-                    |_| PreLockFilterAction::AttemptToSchedule // no pre-lock filter for now
+                    |state| {
+                        if state.last_tried_slot() == Some(bank_start.working_bank.slot()) {
+                            PreLockFilterAction::SkipAndRetain
```

Comment:
to clarify, what are the cases `worker` would send back tx for retry? Wondering if *all* retry cases are not suitable for current slot